monitor.go

```
@@ -0,0 +1,302 @@
+					Data:  chain,
+					Error: err,
+				}
+				monitor.eventCh <- event
+			case unix.NFT_MSG_NEWRULE, unix.NFT_MSG_DELRULE:
+				rule, err := parseRuleFromMsg(msg)
+				event := &Event{
+					Type:  EventType(msgType),
+					Data:  rule,
+					Error: err,
+				}
+				monitor.eventCh <- event
+			case unix.NFT_MSG_NEWSET, unix.NFT_MSG_DELSET:
+				set, err := setsFromMsg(msg)
+				event := &Event{
+					Type:  EventType(msgType),
+					Data:  set,
+					Error: err,
+				}
+				monitor.eventCh <- event
+			case unix.NFT_MSG_NEWSETELEM, unix.NFT_MSG_DELSETELEM:
+				elems, err := elementsFromMsg(uint8(TableFamilyUnspecified), msg)
+				event := &Event{
+					Type:  EventType(msgType),
+					Data:  elems,
+					Error: err,
+				}
+				monitor.eventCh <- event
+			case unix.NFT_MSG_NEWOBJ, unix.NFT_MSG_DELOBJ:
+				obj, err := objFromMsg(msg)
+				event := &Event{
+					Type:  EventType(msgType),
+					Data:  obj,
+					Error: err,
+				}
+				monitor.eventCh <- event
+			case unix.NFT_MSG_TRACE:
+			}
+		}
+	}
+	monitor.mtx.Lock()
+	if monitor.status != monitorClosed {
+		monitor.status = monitorClosed
+		monitor.closer()
+		close(monitor.eventCh)
+	}
+	monitor.mtx.Unlock()
+}
+
+func (monitor *Monitor) Close() {
+	monitor.mtx.Lock()
+	if monitor.status != monitorClosed {
+		monitor.status = monitorClosed
+		monitor.closer()
+		close(monitor.eventCh)
+	}
+	monitor.mtx.Unlock()
+}
+
+// AddMonitor to perform the monitor immediately. The channel will be closed after
+// calling Close on Monitor or encountering a netlink conn error while Receive.
+func (cc *Conn) AddMonitor(monitor *Monitor) (chan *Event, error) {
+	conn, closer, err := cc.netlinkConn()
+	if err != nil {
+		return nil, err
+	}
+	monitor.conn = conn
+	monitor.closer = closer
+
+	if monitor.monitorFlags != 0 {
+		err = conn.JoinGroup(uint32(unix.NFNLGRP_NFTABLES))
+		if err != nil {
+			monitor.closer()
+			return nil, err
+		}
+		conn.JoinGroup(uint32(unix.NFNLGRP_NFTRACE))
```

Why are we ignoring the error here?